apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/PoolRewrite/SessionPool.cs

```diff
@@ -64,6 +66,85 @@ public SessionPool(SpannerClient client, SessionPoolOptions options, Logger logg
             _logger = logger ?? Logger.DefaultLogger;
             _detachedSessionPool = new DetachedSessionPool(this);
             _sessionAcquisitionSemaphore = new SemaphoreSlim(Options.MaximumConcurrentSessionCreates);
+            Task.Run(() => PoolMaintenanceLoop(this));
+        }
+
+        /// <summary>
+        /// A long-running loop performing pool maintenance. Each iteration runs <see cref="MaintainPool"/>.
+        /// This is a static method to allow the target pool to be garbage collected, at which point the 
+        /// </summary>
+        /// <param name="pool">The pool to maintain.</param>
+        /// <returns></returns>
+        private static async Task PoolMaintenanceLoop(SessionPool pool)
+        {
+            // Keep a weak reference so that the pool can be garbage collected and we can stop the
+            // maintenance task.
+            var weakRef = new WeakReference(pool);
+            // Make sure that even if the pool variable is captured due to compiler implementation details,
```

Comment:
nit: Can't you just pass the week reference as the parameter to this method? Althouhg it's cleaner for calling code as it is now.

Reply:
It would then have to be a static method within TargetedSessionPool, which feels a little odd. Basically I wanted TargetedSessionPool to feel as natural as possible, so the WeakReference part is isolated to just this method.

Reply:
Yep, actually I think I prefer leaving all the nulling and garbage collection related stuff isolated here.